### PR TITLE
Backport changes

### PR DESCRIPTION
**Pull request by sgliner-ledger**

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -58,4 +87,5 @@ jobs:
 
       - name: Run tests
         run: |
-          cd tests && pip install -r requirements.txt && SPECULOS=/speculos/speculos.py pytest
+          cd tests && pip install -r requirements.txt && SPECULOS=/speculos/speculos.py pytest --model=${{ matrix.model }}
```
Comment (agrojean-ledger) [inline .github/workflows/ci-workflow.yml:90]: Not possible to use the tests reusable workflow here ? it does more or less the same stuff :
```
pip install --extra-index-url https://test.pypi.org/simple/ -r ${{ inputs.test_dir }}/requirements.txt
pytest ${{ inputs.test_dir }}/ --tb=short -v --device ${{ matrix.device }} -k ${{ inputs.test_filter }}
```

File: Makefile [hunk 2]
```diff
@@ -150,11 +153,17 @@ LDLIBS   += -lm -lgcc -lc
 include $(BOLOS_SDK)/Makefile.glyphs
 
 ### variables processed by the common makefile.rules of the SDK to grab source files and include dirs
-APP_SOURCE_PATH  += src
-SDK_SOURCE_PATH  += lib_stusb lib_stusb_impl lib_u2f qrcode
-SDK_SOURCE_PATH  += lib_ux
+APP_SOURCE_PATH  += src 
+
+APP_SOURCE_FILES += ${BOLOS_SDK}/lib_standard_app/format.c
```
Comment (agrojean-ledger) [inline Makefile:158]: You could simplify the makefile by using the standard one like boilerplate, no ?

File: src/main.c [hunk 3]
```diff
@@ -725,17 +72,22 @@ unsigned short io_exchange_al(unsigned char channel, unsigned short tx_len) {
 }
 
 unsigned char io_event(unsigned char channel) {
```
Comment (agrojean-ledger) [inline src/main.c:74]: You could use the app standard function here couldn't you ?

File: tests/test_get_pubkey.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from bitcoin_client.bitcoin_base_cmd import AddrType
+from utils import automation
+
+
+@automation("automations/accept_pubkey.json")
+def test_get_public_key(cmd):
```
Comment (agrojean-ledger) [inline tests/test_get_pubkey.py:6]: No ragger tests so no way to tell if the UI displays stuff properly : /

File: src/ui_nbgl.c [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+    prompt_cancel(MESSAGE_TYPE); 
+}
+
+static void prompt_cancel_transaction(void) {
+    prompt_cancel(TRANSACTION_TYPE);
+}
+
+static void transaction_review_callback(bool token) {
+  if (token) {
+    releaseContext();
```
Comment (agrojean-ledger) [inline src/ui_nbgl.c:162]: `releaseContext` function still needed ?

File: src/ui_nbgl.c [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+                                    .quitToken = 0,
+                                    .tuneId = TUNE_TAP_CASUAL};
+
+  nbgl_pageContent_t content = {
+      .type = TAG_VALUE_LIST,
+      .tagValueList.nbPairs = uiContext.nbPairs,
+      .tagValueList.pairs = (nbgl_layoutTagValue_t *)uiContext.tagValues
+  };
+
+  pageContext = nbgl_pageDrawGenericContent(&transaction_fee_callback, &info, &content);
```
Comment (agrojean-ledger) [inline src/ui_nbgl.c:373]: You cannot replace `nbgl_pageDrawGenericContent` calls with some `useCase` high level API function ?

File: src/main.c (hunk 3 above)
Comment (sgliner-ledger) [reply src/main.c:74]: I could but I tried to limit the scope to porting related stuff :)

File: Makefile (hunk 2 above)
Comment (sgliner-ledger) [reply Makefile:158]: I could but I tried to limit the scope to porting related stuff :)

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (sgliner-ledger) [reply .github/workflows/ci-workflow.yml:90]: At some point i'll switch the tests to ragger, I'll do that then :)

File: src/ui_nbgl.c (hunk 5 above)
Comment (sgliner-ledger) [reply src/ui_nbgl.c:162]: I'm using the `nbgl_pageDrawGenericContent`, so  I guess yes

**Pull request merged**